frontend/apps/marketing/Dockerfile

```diff
@@ -0,0 +1,82 @@
+# needed for the target docker image.                                        #
+#                                                                            #
+# This generates a partial monorepo removing unneeded packages and apps.     #
+#                                                                            #
+# Input: The base operating system                                           #
+# Output: The pruned partial monorepo for the target app                     #
+##############################################################################
+
+FROM base AS source-code
+
+# Enable corepack
+RUN corepack enable
+
+WORKDIR /app
+
+COPY . .
+ 
+# Generate a partial monorepo with a pruned lockfile for a target workspace.
+# See: https://turbo.build/repo/docs/reference/prune
+RUN yarn turbo prune @code-dot-org/marketing --docker
+
+##############################################################################
+# Builder Layer                                                              #
+#                                                                            #
+# Builds the partial `frontend` monorepo for the target application          #
+#                                                                            #
+# Inputs: The base operating system and pruned `frontend` monorepo           #
+# Output: The built version of the target application                        #
+##############################################################################
+
+FROM base AS builder
+
+# Enable corepack
+RUN corepack enable
+
+WORKDIR /app
+
+# First install the dependencies (as they change less often)
+COPY --from=source-code /app/out/json/ .
+RUN yarn install
+ 
+# Build the project
+COPY --from=source-code /app/out/full/ .
+RUN yarn turbo run build
+
+##############################################################################
+# Runnner Layer                                                              #
+#                                                                            #
+# Runtime execution options for the application                              #
+# Note: the source code is not included in the final image                   #
+#                                                                            #
+# Inputs: The base operating system and built version of the target app      #
+# Output: Rootless execution env for the target app with minimal artifacts   #
+##############################################################################
+
+FROM base AS runner
+WORKDIR /app
+ 
+# Don't run production as root
+RUN addgroup --system --gid 1001 nodejs
+RUN adduser --system --uid 1001 nextjs
+USER nextjs
+ 
+# Automatically leverage output traces to reduce image size
+# https://nextjs.org/docs/advanced-features/output-file-tracing
+COPY --from=builder --chown=nextjs:nodejs /app/apps/marketing/.next/standalone ./
+COPY --from=builder --chown=nextjs:nodejs /app/apps/marketing/.next/static ./apps/marketing/.next/static
```

Comment:
You might be interested in using `--link` if this is a significant amount of data or a large number of files (e.g. an node_modules dir). --link will link the layer in from builder rather than copying the files, its effectively an instant op. Not important by any means, but might be a "nice to know".

Reply:
See: https://docs.docker.com/reference/dockerfile/#copy---link

You have to have opted into:
```
# syntax=docker/dockerfile:1.4
```

Reply:
TIL, though these are extremely small build outputs so this is a very minor optimization. That being said, I don't see a reason why we shouldn't add it.